Do a VM guest shut down if possible (#720)
harvester/docs: docs/advanced/addons/vmimport.md

    namespace: default
    kind: VmwareSource
    apiVersion: migration.harvesterhci.io/v1beta1
  gracefulShutdown: true
  gracefulShutdownTimeout: 300000000000
```

> This will trigger the controller to export the VM named "alpine-export-test" on the VMware source cluster to be exported, processed and recreated into the harvester cluster

Reviewer: ```suggestion
This will trigger the controller to export the VM named "alpine-export-test" on the VMware source cluster to be exported, processed and recreated into the Harvester cluster
```

Author: Done incl. the versioned docs as well.

This can take a while based on the size of the virtual machine, but users should see `VirtualMachineImages` created for each disk in the defined virtual machine.

The list of items in `networkMapping` will define how the source network interfaces are mapped to the Harvester Networks.

If a match is not found, each unmatched network interface is attached to the default `managementNetwork`.

By default, the VM is powered off. If you want the guest OS to gracefully shutdown the VM,
you can set the `gracefulShutdown` field to `true`. Additionally, you can set an optional
timeout with the `gracefulShutdownTimeout` field which is in nanoseconds. It defaults to
1 minute. If the VM is not gracefully shutdown within the configured time period, then a 

> hard power off is enforced.

Author: Done

> - Harvester enforces a hard power off if the virtual machine is not gracefully shut down within the configured period.

Reviewer 2: ```suggestion
Harvester enforces a hard power off if the virtual machine is not gracefully shut down within the configured period.
```

Reviewer 2: ![Screenshot 2025-04-01 at 12 37 39 PM](https://github.com/user-attachments/assets/6e2d7f48-2c24-4925-bb31-1d1633f2d0ac)

Reviewer 2: Please remove the hyphen. That sentence is not part of the list.

Author: Done

:::note
The `gracefulShutdown` field is not supported by the OpenStack VM importer because a

> graceful shutdown is done automatically by OpenStack.

Author: Done

The `storageClass` field specifies the [StorageClass](../storageclass.md) to be used for images and provisioning persistent volumes during the import process. If not specified, the default StorageClass will be used.

> The virtual machine is stopped by default by the guest operating system shutting it down properly before the import is started. Harvester enforces a hard power off if the virtual machine is not gracefully shut down within the configured period. This period can be modified by setting the `gracefulShutdownTimeout` field. The value is specified in nanoseconds and is set to `60000000000` (one minute) by default. To deactivate the graceful shutdown and thus force a hard power off set `gracefulShutdown` to `false`.

Reviewer 2: ```suggestion
By default, the guest operating system stops and shuts down the virtual machine before the import is started. If the virtual machine is not gracefully shut down within a specific period, Harvester forces a hard power off. You can adjust this shut down period by modifying the value of the `gracefulShutdownTimeout` field, which is set to `60000000000` nanoseconds (equivalent to one minute) by default. If you want to disable the graceful shutdown and force a hard power off, set the value of the `gracefulShutdown` field to `false`.
```

Author: Done
